src/typing.py

```diff
@@ -1985,20 +1987,29 @@ class Employee(NamedTuple):
         The resulting class has one extra attribute: _field_types,
         giving a dict mapping field names to types.  (The field names
         are in the _fields attribute, which is part of the namedtuple
-        API.) Backward-compatible usage::
+        API.) Alternative equivalent keyword syntax is also accepted::
+
+            Employee = NamedTuple('Employee', name=str, id=int)
+
+        Backward-compatible usage::
 
             Employee = NamedTuple('Employee', [('name', str), ('id', int)])
         """
 
-        def __new__(self, typename, fields):
+        def __new__(self, typename, fields=None, **kwargs):
+            if fields is None:
+                fields = kwargs.items()
```

Comment:
This should be disallowed for Python <= 3.5. (Remember that this code will be in the "typing" package which should work for Python 3.2 and up, and it will be copied verbatim into the 3.5 branch as well.)